pagerduty/resource_pagerduty_event_orchestration_integration.go

```diff
@@ -0,0 +1,309 @@
+		return nil
+	})
+
+	if retryErr != nil {
+		time.Sleep(2 * time.Second)
+		return diag.FromErr(retryErr)
+	}
+
+	return nil
+}
+
+func resourcePagerDutyEventOrchestrationIntegrationRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client, err := meta.(*Config).Client()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	id := d.Id()
+	oid := d.Get("event_orchestration").(string)
+
+	retryErr := resource.RetryContext(ctx, 2*time.Minute, func() *resource.RetryError {
+		log.Printf("[INFO] Reading Integration '%s' for PagerDuty Event Orchestration: %s", id, oid)
+
+		if integration, _, err := client.EventOrchestrationIntegrations.GetContext(ctx, oid, id); err != nil {
+			time.Sleep(2 * time.Second)
+			return resource.RetryableError(err)
+		} else if integration != nil {
+			setEventOrchestrationIntegrationProps(d, integration)
+		}
+		return nil
+	})
+
+	if retryErr != nil {
+		return diag.FromErr(retryErr)
+	}
+
+	return nil
+}
+
+func resourcePagerDutyEventOrchestrationIntegrationUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client, err := meta.(*Config).Client()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	id := d.Id()
+
+	// Migrate integration if the event_orchestration property was modified
+	if d.HasChange("event_orchestration") {
+		o, n := d.GetChange("event_orchestration")
+		sourceOrchId := o.(string)
+		destinationOrchId := n.(string)
+
+		retryErr := resource.RetryContext(ctx, 2*time.Minute, func() *resource.RetryError {
+			log.Printf("[INFO] Migrating Event Orchestration Integration '%s': source - '%s', destination - '%s'", id, sourceOrchId, destinationOrchId)
+
+			if _, _, err := client.EventOrchestrationIntegrations.MigrateFromOrchestrationContext(ctx, destinationOrchId, sourceOrchId, id); err != nil {
+				if isErrCode(err, 400) {
+					return resource.NonRetryableError(err)
+				}
+				return resource.RetryableError(err)
+			} else {
+				// try reading the migrated integration from destination and source:
+				_, _, readDestErr := client.EventOrchestrationIntegrations.GetContext(ctx, destinationOrchId, id)
+				srcInt, _, readSrcErr := client.EventOrchestrationIntegrations.GetContext(ctx, sourceOrchId, id)
+
+				// retry migration if the read request returned an error:
+				if readDestErr != nil {
+					log.Printf("[WARN] Integration '%s' cannot be found on the destination PagerDuty Event Orchestration '%s'. Retrying migration....", id, destinationOrchId)
+					return resource.RetryableError(readDestErr)
+				}
+
+				// retry migration if the integration still exists on the source:
+				if readSrcErr == nil && srcInt != nil {
+					log.Printf("[WARN] Integration '%s' still exists on the source PagerDuty Event Orchestration '%s'. Retrying migration....", id, sourceOrchId)
+					return resource.RetryableError(fmt.Errorf("Integration '%s' still exists on the source PagerDuty Event Orchestration '%s'.", id, sourceOrchId))
+				}
```

Comment:
TODO: refactor to a separate functions with retries (see pagerduty/resource_pagerduty_team_membership.go)

Reply:
I used `.GetContext()` here directly instead of calling `fetchPagerDutyEventOrchestrationIntegration()` because when we perform a migration we don't need to set any integration properties that the fetch function does, especially when we are verifying that the integration doesn't exist on the source.